docs/zh-CN/docs/sql-manual/sql-functions/array-functions/array_filter.md

```diff
@@ -0,0 +1,95 @@
+---
+{
+    "title": "array_filter",
+    "language": "zh-CN"
+}
+---
+
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+## array_filter
+
+<version since="1.2.2">
+
+array_filter(lambda,array)
+
+</version>
+
+### description
+
+Use a lambda expression as the input parameter to calculate and filter the internal data of other input ARRAY parameters.
+Filter out the values of 0 and NULL in the result, and only keep the values of 1.
+
+```
+array_filter(x->x>0, array1);
+array_filter(x->(x+2)=10, array1);
+array_filter(x->(abs(x)-2)>0, array1);
```

Comment:
what will occur if the first lambda function do not return `boolean`? 
1. return error ?
2. do cast to boolean?

Reply:
have a cast to bool